platform/web/detect.py

```diff
@@ -259,37 +259,41 @@ def configure(env: "SConsEnvironment"):
 
     # Run the main application in a web worker
     if env["proxy_to_pthread"]:
-        env.Append(LINKFLAGS=["-s", "PROXY_TO_PTHREAD=1"])
+        env.Append(LINKFLAGS=["-sPROXY_TO_PTHREAD=1"])
         env.Append(CPPDEFINES=["PROXY_TO_PTHREAD_ENABLED"])
-        env.Append(LINKFLAGS=["-s", "EXPORTED_RUNTIME_METHODS=['_emscripten_proxy_main']"])
+        env.Append(LINKFLAGS=["-sEXPORTED_RUNTIME_METHODS=['_emscripten_proxy_main']"])
         # https://github.com/emscripten-core/emscripten/issues/18034#issuecomment-1277561925
-        env.Append(LINKFLAGS=["-s", "TEXTDECODER=0"])
+        env.Append(LINKFLAGS=["-sTEXTDECODER=0"])
         # BigInt support to pass object pointers between contexts
         needs_wasm_bigint = True
 
     if needs_wasm_bigint:
-        env.Append(LINKFLAGS=["-s", "WASM_BIGINT"])
+        env.Append(LINKFLAGS=["-sWASM_BIGINT"])
 
     # Reduce code size by generating less support code (e.g. skip NodeJS support).
-    env.Append(LINKFLAGS=["-s", "ENVIRONMENT=web,worker"])
+    env.Append(LINKFLAGS=["-sENVIRONMENT=web,worker"])
 
     # Wrap the JavaScript support code around a closure named Godot.
-    env.Append(LINKFLAGS=["-s", "MODULARIZE=1", "-s", "EXPORT_NAME='Godot'"])
+    env.Append(LINKFLAGS=["-sMODULARIZE=1", "-sEXPORT_NAME='Godot'"])
+
+    # Force long jump mode to 'wasm'
+    env.Append(CCFLAGS=["-sSUPPORT_LONGJMP='wasm'"])
+    env.Append(LINKFLAGS=["-sSUPPORT_LONGJMP='wasm'"])
```

Comment:
I would add an emscripten version check here. It's a new feature, right?

Reply:
[No, not at all.](https://github.com/emscripten-core/emscripten/blob/main/ChangeLog.md#v13610-9242016)